e2e/cypress/tests/support/db_commands.ts

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {ChainableT} from '../types';
+
+const dbClient = Cypress.env('dbClient');
+const dbConnection = Cypress.env('dbConnection');
+const dbConfig = {
+    client: dbClient,
+    connection: dbConnection,
+};
+
+const message = `Compare "cypress.json" against "config.json" of mattermost-server. It should match database driver and connection string.
+
+The value at "cypress.json" is based on default mattermost-server's local database: 
+{"dbClient": "${dbClient}", "dbConnection": "${dbConnection}"}
+
+If your server is using database other than the default, you may export those as env variables, like:
+"__CYPRESS_dbClient=[dbClient] CYPRESS_dbConnection=[dbConnection] npm run cypress:open__"
+`;
+
+function apiRequireServerDBToMatch(): ChainableT {
+    return cy.apiGetConfig().then(({config}) => {
+        // On Cloud, SqlSettings is not being returned.
+        // With that, checking of server DB will be ignored and will assume it does match with
+        // the one being expected by Cypress.
+        if (config.SqlSettings && config.SqlSettings.DriverName !== dbClient) {
+            expect(config.SqlSettings.DriverName, message).to.equal(dbClient);
+        }
+    });
+}
+Cypress.Commands.add('apiRequireServerDBToMatch', apiRequireServerDBToMatch);
+
+/**
+ * Gets active sessions of a user on a given username or user ID directly from the database
+ * @param {String} username
+ * @param {String} userId
+ * @param {String} limit - maximum number of active sessions to return, e.g. 50 (default)
+ * @returns {Object} user - user object
+ * @returns {[Object]} sessions - an array of active sessions
+ */
+function dbGetActiveUserSessions({username, userId, limit}): ChainableT<{user: Record<string, any>; sessions: Array<Record<string, any>>}> {
+    return cy.task('dbGetActiveUserSessions', {dbConfig, params: {username, userId, limit}}).then(({user, sessions, errorMessage}) => {
+        expect(errorMessage).to.be.undefined;
+
+        return cy.wrap({user, sessions});
+    });
+}
+Cypress.Commands.add('dbGetActiveUserSessions', dbGetActiveUserSessions);
```

Comment:
I suggest to follow the pattern we have for commands in support file, e.g. [apiInitSetup](https://github.com/mattermost/mattermost-webapp/blob/master/e2e/cypress/tests/support/api/setup.ts) where function parameter uses `interface` for types (instead of JSON notation) and then JSON notation is declared under `Cypress.Chainable` namespace.
Please avoid using `Record` whenever types is already available, like `user` for `Cypress.UserProfile`, `session` for `Cypress.Session`.

Please do the same for other commands in this support file.

Reply:
I updated the code according to your example.
I updated to use `Cypress.UserProfile` type for the user, but I was unable to use the `Cypress.Session` type since it's not consistent with type expected by the tests. The tests use fields like `createat` and `userid` while the fields in the `Cypress.Session` type are defined as `create_at` and `user_id`.
Let me know if you want me to do anything else here.

Reply:
Oh I see. Also `Cypress.Session` conflicts with the built-in interface of Cypress itself. Maybe one option is to deal with the type conversion at `db_request.js` both for User and Session; a good time would be once it's migrated to TS. I'm good as is but let me know if you'd like to take a shot here or in a separate PR.